File: suc/kotlin/007.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "010_ShoppingCart_en"
private const val UI_TYPE = "E-commerce Cart"
private const val STYLE_THEME = "Minimal Monochrome"
private const val LANG = "en"
private const val BASELINE_SIZE = "1280x720"

object AppTokens {
    object Colors {
        val primary = Color(0xFF000000)
        val secondary = Color(0xFFFF6F00)
        val tertiary = Color(0xFF4CAF50)
        val background = Color(0xFFFFFFFF)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFF5F5F5)
        val outline = Color(0xFFE0E0E0)
        val success = Color(0xFF4CAF50)
        val warning = Color(0xFFFF9800)
        val error = Color(0xFFF44336)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF000000)
        val onSurface = Color(0xFF000000)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 22.sp, fontWeight = FontWeight.SemiBold, lineHeight = 28.sp, letterSpacing = 0.sp)
        val headline = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.SemiBold, lineHeight = 24.sp, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium, lineHeight = 18.sp, letterSpacing = 0.sp)
        val body = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Normal, lineHeight = 16.sp, letterSpacing = 0.sp)
        val label = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Medium, lineHeight = 14.sp, letterSpacing = 0.sp)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }
    object Spacing {
        val xs = 2.dp
        val sm = 4.dp
        val md = 6.dp
        val lg = 8.dp
        val xl = 12.dp
        val xxl = 16.dp
        val xxxl = 24.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 2.dp, 0.5.dp, 0.12f)
        val level2 = ShadowSpec(2.dp, 4.dp, 1.dp, 0.14f)
        val level3 = ShadowSpec(3.dp, 6.dp, 1.5.dp, 0.16f)
        val level4 = ShadowSpec(4.dp, 8.dp, 2.dp, 0.18f)
        val level5 = ShadowSpec(5.dp, 10.dp, 2.5.dp, 0.20f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            extraSmall = AppTokens.Shapes.small,
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large,
            extraLarge = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class CartItem(val id: Int, val name: String, val price: Double, val color: Color)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val cartItems = remember {
        listOf(
            CartItem(1, "Wireless Headphones", 129.99, AppTokens.Colors.secondary),
            CartItem(2, "Smartphone Case", 24.99, AppTokens.Colors.primary),
            CartItem(3, "USB-C Cable", 19.99, AppTokens.Colors.tertiary),
            CartItem(4, "Portable Charger", 49.99, AppTokens.Colors.secondary)
        )
    }
    val quantities = remember { mutableStateListOf(1, 1, 1, 1) }
    val subtotal = remember { mutableStateOf(0.0) }
    val shipping = 5.99
    val tax = 18.50

    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)) {
                        Box(modifier = Modifier.size(20.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                        Text(text = "Shopping Cart", style = MaterialTheme.typography.displayLarge, color = MaterialTheme.colorScheme.onSurface)
                    }
                }
            )
        },
        bottomBar = {
            Surface(color = MaterialTheme.colorScheme.surface, tonalElevation = AppTokens.ElevationMapping.level3.elevation, shadowElevation = AppTokens.ElevationMapping.level3.elevation) {
                Column(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)) {
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                        Text(text = "Subtotal", style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = "$${"%.2f".format(cartItems.mapIndexed { index, item -> item.price * quantities[index] }.sum())}", style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                        Text(text = "Shipping", style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = "$${"%.2f".format(shipping)}", style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                        Text(text = "Tax", style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = "$${"%.2f".format(tax)}", style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                        Text(text = "Total", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = "$${"%.2f".format(cartItems.mapIndexed { index, item -> item.price * quantities[index] }.sum() + shipping + tax)}", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Button(onClick = {}, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.primary, contentColor = MaterialTheme.colorScheme.onPrimary), shape = AppTokens.Shapes.large, modifier = Modifier.fillMaxWidth().height(44.dp)) {
                        Text(text = "Proceed to Checkout", style = MaterialTheme.typography.titleMedium)
                    }
                }
            }
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { padding ->
        LazyColumn(modifier = Modifier.fillMaxSize().padding(padding).padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg), contentPadding = PaddingValues(bottom = AppTokens.Spacing.xxxl)) {
            items(cartItems) { item ->
                Card(shape = AppTokens.Shapes.large, colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface), elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation), border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), modifier = Modifier.fillMaxWidth().heightIn(min = 100.dp)) {
                    Row(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg), verticalAlignment = Alignment.CenterVertically) {
                        Box(modifier = Modifier.size(60.dp).background(item.color, AppTokens.Shapes.medium))
                        Column(modifier = Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs), horizontalAlignment = Alignment.Start) {
                            Text(text = item.name, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                            Text(text = "$${"%.2f".format(item.price)}", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                            Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                                Box(modifier = Modifier.size(28.dp).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.small))
                                Box(modifier = Modifier.size(28.dp).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.small))
                                Box(modifier = Modifier.size(28.dp).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.small))
                            }
                        }
                        Column(verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), horizontalAlignment = Alignment.CenterHorizontally) {
                            Button(onClick = { if (quantities[item.id - 1] < 10) quantities[item.id - 1]++ }, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.surfaceVariant, contentColor = MaterialTheme.colorScheme.onSurface), shape = AppTokens.Shapes.small, modifier = Modifier.size(28.dp)) {
                                Text(text = "+", style = MaterialTheme.typography.titleMedium)
                            }
                            Text(text = quantities[item.id - 1].toString(), style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                            Button(onClick = { if (quantities[item.id - 1] > 1) quantities[item.id - 1]-- }, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.surfaceVariant, contentColor = MaterialTheme.colorScheme.onSurface), shape = AppTokens.Shapes.small, modifier = Modifier.size(28.dp)) {
                                Text(text = "-", style = MaterialTheme.typography.titleMedium)
                            }
                        }
                        Button(onClick = {}, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.error, contentColor = MaterialTheme.colorScheme.onPrimary), shape = AppTokens.Shapes.medium, modifier = Modifier.height(36.dp).width(85.dp)) {
                            Text(text = "Remove", style = MaterialTheme.typography.labelMedium)
                        }
                    }
                }
            }
            item {
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                    AssistChip(onClick = {}, label = { Text(text = "Apply Coupon", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.surface))
                    AssistChip(onClick = {}, label = { Text(text = "Gift Wrap", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.surface))
                    AssistChip(onClick = {}, label = { Text(text = "Save for Later", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.surface))
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.navigationBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.navigationBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFFFFFFF, widthDp = 360, heightDp = 640)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/007.swift
import SwiftUI

enum ShoppingCartTokens {
    enum Colors {
        static let primary = Color.black
        static let secondary = Color(red: 255 / 255, green: 111 / 255, blue: 0 / 255)
        static let tertiary = Color(red: 76 / 255, green: 175 / 255, blue: 80 / 255)
        static let background = Color.white
        static let surface = Color.white
        static let surfaceVariant = Color(red: 245 / 255, green: 245 / 255, blue: 245 / 255)
        static let outline = Color(red: 224 / 255, green: 224 / 255, blue: 224 / 255)
        static let error = Color(red: 244 / 255, green: 67 / 255, blue: 54 / 255)
        static let onPrimary = Color.white
        static let onBackground = Color.black
        static let onSurface = Color.black
    }

    enum Typography {
        static let display = Font.system(size: 22, weight: .semibold)
        static let headline = Font.system(size: 18, weight: .semibold)
        static let title = Font.system(size: 14, weight: .medium)
        static let body = Font.system(size: 12, weight: .regular)
        static let label = Font.system(size: 10, weight: .medium)
    }

    enum Shapes {
        static let small: CGFloat = 6
        static let medium: CGFloat = 8
        static let large: CGFloat = 12
    }

    enum Spacing {
        static let xs: CGFloat = 2
        static let sm: CGFloat = 4
        static let md: CGFloat = 6
        static let lg: CGFloat = 8
        static let xl: CGFloat = 12
        static let xxl: CGFloat = 16
        static let xxxl: CGFloat = 24
    }

    struct ShadowSpec {
        let radius: CGFloat
        let y: CGFloat
        let opacity: Double
    }

    enum Elevation {
        static let level1 = ShadowSpec(radius: 2, y: 0.5, opacity: 0.12)
        static let level2 = ShadowSpec(radius: 4, y: 1, opacity: 0.14)
        static let level3 = ShadowSpec(radius: 6, y: 1.5, opacity: 0.16)
        static let level4 = ShadowSpec(radius: 8, y: 2, opacity: 0.18)
        static let level5 = ShadowSpec(radius: 10, y: 2.5, opacity: 0.2)
    }
}

struct CartItem: Identifiable {
    let id: Int
    let name: String
    let price: Double
    let color: Color
}

struct ShoppingCartAppView: View {
    private let cartItems: [CartItem] = [
        CartItem(id: 1, name: "Wireless Headphones", price: 129.99, color: ShoppingCartTokens.Colors.secondary),
        CartItem(id: 2, name: "Smartphone Case", price: 24.99, color: ShoppingCartTokens.Colors.primary),
        CartItem(id: 3, name: "USB-C Cable", price: 19.99, color: ShoppingCartTokens.Colors.tertiary),
        CartItem(id: 4, name: "Portable Charger", price: 49.99, color: ShoppingCartTokens.Colors.secondary)
    ]

    @State private var quantities: [Int] = [1, 1, 1, 1]

    private let shipping: Double = 5.99
    private let tax: Double = 18.50
    private let bottomBarHeight: CGFloat = 220

    private var subtotal: Double {
        zip(cartItems, quantities).map { item, quantity in
            item.price * Double(quantity)
        }
        .reduce(0, +)
    }

    private var total: Double {
        subtotal + shipping + tax
    }

    var body: some View {
        ZStack(alignment: .bottom) {
            ShoppingCartTokens.Colors.background
                .ignoresSafeArea()

            VStack(spacing: 0) {
                topBar
                    .padding(.top, 30)
                    .padding(.bottom, 20)
                ScrollView(showsIndicators: false) {
                    LazyVStack(spacing: ShoppingCartTokens.Spacing.lg) {
                        ForEach(Array(cartItems.enumerated()), id: \.element.id) { index, item in
                            cartItemCard(item: item, index: index)
                        }
                        chipsRow
                    }
                    .padding(.horizontal, ShoppingCartTokens.Spacing.lg)
                    .padding(.top, ShoppingCartTokens.Spacing.md)
                    .padding(.bottom, bottomBarHeight + ShoppingCartTokens.Spacing.lg)
                }
            }

            bottomSummary
                .padding(.horizontal, ShoppingCartTokens.Spacing.lg)
                .padding(.bottom, ShoppingCartTokens.Spacing.md)
        }
        .statusBarHidden(true)
    }

    private var topBar: some View {
        HStack(spacing: ShoppingCartTokens.Spacing.sm) {
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.small, style: .continuous)
                .fill(ShoppingCartTokens.Colors.primary)
                .frame(width: 20, height: 20)
            Text("Shopping Cart")
                .font(ShoppingCartTokens.Typography.display)
                .foregroundColor(ShoppingCartTokens.Colors.onSurface)
        }
        .frame(maxWidth: .infinity)
        .padding(.top, ShoppingCartTokens.Spacing.xxl)
        .padding(.bottom, ShoppingCartTokens.Spacing.md)
    }

    private func cartItemCard(item: CartItem, index: Int) -> some View {
        HStack(spacing: ShoppingCartTokens.Spacing.lg) {
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.medium, style: .continuous)
                .fill(item.color)
                .frame(width: 60, height: 60)

            VStack(alignment: .leading, spacing: ShoppingCartTokens.Spacing.xs) {
                Text(item.name)
                    .font(ShoppingCartTokens.Typography.title)
                    .foregroundColor(ShoppingCartTokens.Colors.onSurface)
                    .frame(maxWidth: 120, alignment: .leading)
                    .lineLimit(2)

                Text(String(format: "$%.2f", item.price))
                    .font(ShoppingCartTokens.Typography.headline)
                    .foregroundColor(ShoppingCartTokens.Colors.onSurface)

                HStack(spacing: ShoppingCartTokens.Spacing.sm) {
                    featureBox
                    featureBox
                    featureBox
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            VStack(spacing: ShoppingCartTokens.Spacing.sm) {
                quantityButton(symbol: "+") {
                    if quantities.indices.contains(index), quantities[index] < 10 {
                        quantities[index] += 1
                    }
                }
                Text("\(quantities.indices.contains(index) ? quantities[index] : 1)")
                    .font(ShoppingCartTokens.Typography.title)
                    .foregroundColor(ShoppingCartTokens.Colors.onSurface)
                quantityButton(symbol: "-") {
                    if quantities.indices.contains(index), quantities[index] > 1 {
                        quantities[index] -= 1
                    }
                }
            }

            Button(action: {}) {
                Text("Remove")
                    .font(ShoppingCartTokens.Typography.label)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(width: 85, height: 36)
            .background(
                RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.medium, style: .continuous)
                    .fill(ShoppingCartTokens.Colors.error)
            )
            .foregroundColor(ShoppingCartTokens.Colors.onPrimary)
        }
        .padding(ShoppingCartTokens.Spacing.lg)
        .frame(maxWidth: .infinity, minHeight: 100, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.large, style: .continuous)
                .fill(ShoppingCartTokens.Colors.surface)
        )
        .overlay(
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.large, style: .continuous)
                .stroke(ShoppingCartTokens.Colors.outline, lineWidth: 1)
        )
        .shadow(color: Color.black.opacity(ShoppingCartTokens.Elevation.level2.opacity),
                radius: ShoppingCartTokens.Elevation.level2.radius,
                x: 0,
                y: ShoppingCartTokens.Elevation.level2.y)
    }

    private var featureBox: some View {
        RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.small, style: .continuous)
            .fill(ShoppingCartTokens.Colors.surfaceVariant)
            .frame(width: 28, height: 28)
    }

    private func quantityButton(symbol: String, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Text(symbol)
                .font(ShoppingCartTokens.Typography.title)
                .frame(maxWidth: .infinity, maxHeight: .infinity)
        }
        .frame(width: 28, height: 28)
        .background(
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.small, style: .continuous)
                .fill(ShoppingCartTokens.Colors.surfaceVariant)
        )
        .foregroundColor(ShoppingCartTokens.Colors.onSurface)
    }

    private var chipsRow: some View {
        let chips = ["Apply Coupon", "Gift Wrap", "Save for Later"]
        return HStack(spacing: ShoppingCartTokens.Spacing.sm) {
            ForEach(chips, id: \.self) { label in
                Button(action: {}) {
                    Text(label)
                        .font(ShoppingCartTokens.Typography.label)
                        .padding(.horizontal, ShoppingCartTokens.Spacing.xl)
                        .padding(.vertical, ShoppingCartTokens.Spacing.sm)
                }
                .background(
                    RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.small, style: .continuous)
                        .fill(ShoppingCartTokens.Colors.surface)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.small, style: .continuous)
                        .stroke(ShoppingCartTokens.Colors.outline, lineWidth: 1)
                )
                .foregroundColor(ShoppingCartTokens.Colors.onSurface)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private var bottomSummary: some View {
        VStack(spacing: ShoppingCartTokens.Spacing.md) {
            summaryRow(title: "Subtotal", value: subtotal)
            summaryRow(title: "Shipping", value: shipping)
            summaryRow(title: "Tax", value: tax)
            summaryRow(title: "Total", value: total, emphasize: true)
            Button(action: {}) {
                Text("Proceed to Checkout")
                    .font(ShoppingCartTokens.Typography.title)
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
            }
            .frame(height: 44)
            .background(
                RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.large, style: .continuous)
                    .fill(ShoppingCartTokens.Colors.primary)
            )
            .foregroundColor(ShoppingCartTokens.Colors.onPrimary)
        }
        .padding(ShoppingCartTokens.Spacing.lg)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: ShoppingCartTokens.Shapes.large, style: .continuous)
                .fill(ShoppingCartTokens.Colors.surface)
        )
        .shadow(color: Color.black.opacity(ShoppingCartTokens.Elevation.level3.opacity),
                radius: ShoppingCartTokens.Elevation.level3.radius,
                x: 0,
                y: ShoppingCartTokens.Elevation.level3.y)
    }

    private func summaryRow(title: String, value: Double, emphasize: Bool = false) -> some View {
        HStack {
            Text(title)
                .font(emphasize ? ShoppingCartTokens.Typography.headline : ShoppingCartTokens.Typography.body)
                .foregroundColor(ShoppingCartTokens.Colors.onSurface)
            Spacer()
            Text(String(format: "$%.2f", value))
                .font(emphasize ? ShoppingCartTokens.Typography.headline : ShoppingCartTokens.Typography.title)
                .foregroundColor(ShoppingCartTokens.Colors.onSurface)
        }
    }
}

struct ShoppingCartAppView_Previews: PreviewProvider {
    static var previews: some View {
        ShoppingCartAppView()
    }
}

@main
struct ShoppingCartApp: App {
    var body: some Scene {
        WindowGroup {
            ShoppingCartAppView()
                .statusBarHidden(true)
                .background(ShoppingCartTokens.Colors.background.ignoresSafeArea())
        }
    }
}
